webdriver-spec.html

```diff
@@ -1435,288 +1435,532 @@
-    in <var>unprocessed capabilities</var>.
+  <li><p>Let <var>dc</var> be the entry in <var>desired capabilities</var> at
+  index <var>k</var>.
 
-   <li><p>If during the steps below the <var>unprocessed capability</var>
-    equals an entry in <var>required capabilities</var>
-    and name of <var>unprocessed capability</var> entry among the names
-    of entries in <var>server capabilities</var>
-    and the values do not match do the following:
+  <li><p>If <var>dc</var> is not a JSON Object,
+  return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
 
-    <ol>
-     <li><p>Append a string containing the property name
-      and the differences between the values.
-    </ol>
+  <li><p>Let <var>merged capabilities</var> be the result of
+  <a>merging capabilities</a> with <var>required capabilities</var> and
+  <var>dc</var> as arguments.
 
-   <li><p>Let <var>browser name</var> be the result of <a>getting a property</a>
-    named <code>browserName</code> from <var>unprocessed capability</var>.
-    If <var>browser name</var> is <a>undefined</a> move to the next step.
-
-   <li><p>Let <var>browser version</var> be the result of <a>getting a property</a>
-    named <code>browserVersion</code> from <var>unprocessed capability</var>.
-    If <var>browser version</var> is <a>undefined</a> move to the next step.
-
-   <li><p>Let <var>platform name</var> be the result of <a>getting a property</a>
-    named <code>platformName</code> from <var>unprocessed capability</var>.
-    If <var>platform name</var> is <a>undefined</a> move to the next step.
-
-   <li><p>Let <var>platform version</var> be the result of <a>getting a property</a>
-    named <code>platformVersion</code> from <var>unprocessed capability</var>.
-    If <var>platform version</var> is <a>undefined</a> move to the next step.
-
-   <li><p>Let <var>insecure ssl</var> be the result of <a>getting a property</a>
-    named <code>acceptSslCerts</code> from <var>unprocessed capability</var>.
-
-    <p>If <var>insecure ssl</var> is defined,
-     unset the <a>active session</a>’s <a>secure SSL</a> state.
-     Otherwise move to the next step.
-
-   <li><p>Let <var>proxy</var> be the result of <a>getting a property</a>
-    named <code>proxy</code> from <var>unprocessed capability</var>.
-    If <var>proxy</var> is <a>undefined</a> move to the next step.
-    If <var>proxy</var> is defined and not a map,
-    append a string saying that a JSON Object is required,
-    else call <a>set the proxy</a> passing in <var>proxy</var>.
-
-   <li><p>Let <var>page load strategy</var> be the result of <a>getting a property</a>
-    named <code>pageLoadStrategy</code> from <var>unprocessed capability</var>.
-    If <var>page load strategy</var> is <a>undefined</a>,
-    then set the entry <code>pageLoadStrategy</code>
-    in <var>server capabilities</var> to <a>normal</a>.
-  </ol>
+  <li><p>If <var>merged capabilities</var> is an <a>error</a>,
```

Comment:
This can be merged with the point above to make

```
<li><p>Let <var>merged capabilities</var> be the result of
 <a>merging capabilities</a> with <var>required capabilities</var> and
<var>dc</var> as arguments. 
 <p>Otherwise, return <var>merged capabilties</var>
```